### PR TITLE
en: aggregate multiple TiDB Monitor data by thanos query

### PR DESCRIPTION
**Pull request by Joyinqin**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)
aggregate multiple TiDB Monitor data by thanos query.

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-tidb-operator/pull/640
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+## Thanos
+
+[Thanos](https://thanos.io/design.md/) is a high availability solution for Prometheus to simplify the availability guarantee of Prometheus.
+
+Thanos provides [Thanos Query](https://thanos.io/components/query.md/) component as a unified query solution across Prometheus. You can use this feature to solve the problem of aggregating monitoring data of multiple clusters.
+
+## Configure Thanos Query
+
+First, you need to configure a Thanos Sidecar container for each TidbMonitor. To update TidbMonitor, configure Thanos Sidecar and deploy Thanos Query component, refer to [Example](https://github.com/pingcap/tidb-operator/tree/master/examples/monitor-with-thanos/README.md).In Thanos Query, a Prometheus corresponds to a Store, which corresponds to a TidbMonitor. After deploying Thanos Query, you can provide a unified query interface for monitoring data through Thanos Query's API.
```
Comment (ran-huang) [inline en/aggregate-multiple-cluster-monitor-data.md:18]:
```suggestion
First, you need to configure a Thanos Sidecar container for each TidbMonitor. Update the `TidbMonitor` CR to configure Thanos Sidecar, and deploy the Thanos Query component. For details, refer to [Example](https://github.com/pingcap/tidb-operator/tree/master/examples/monitor-with-thanos/README.md).

In Thanos Query, a Prometheus corresponds to a Store, which corresponds to a TidbMonitor. After deploying Thanos Query, you can provide a unified query interface for monitoring data through Thanos Query's API.
```

Comment (ran-huang) [reply en/aggregate-multiple-cluster-monitor-data.md:18]: @DanielZhangQD Can we incorporate the [example](https://github.com/pingcap/tidb-operator/tree/master/examples/monitor-with-thanos/README.md) into this doc?

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+## Configure Grafana
+
+After you deploy Thanos Query, Grafana only needs to change the DataSource into the Thanos source to query the monitoring data of multiple TidbMonitors.
```
Comment (ran-huang) [inline en/aggregate-multiple-cluster-monitor-data.md:22]:
```suggestion
After you deploy Thanos Query, change Grafana's DataSource into Thanos to query the monitoring data of multiple `TidbMonitor` CRs.
```

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+## Add or reduce TidbMonitor
+
+If you need to update or offline TidbMonitor, update the starting configuration `--store` of Thanos Query Store, and perform a rolling update to Thanos Query component.
```
Comment (ran-huang) [inline en/aggregate-multiple-cluster-monitor-data.md:26]:
```suggestion
If you need to update the `TidbMonitor` CR or take it offline, update the starting configuration `--store` of Thanos Query Store, and perform a rolling update to the Thanos Query component.
```

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+## Configure archives and storage of Thanos Sidecar
+
+Thanos Sidecar supports replicating monitoring data to S3 remote storage, and the configuration is as follows:
+
+The configuration of the TidbMonitor CR is as follows:
```
Comment (ran-huang) [inline en/aggregate-multiple-cluster-monitor-data.md:32]:
```suggestion
Thanos Sidecar supports replicating monitoring data to S3 remote storage.

The configuration of the `TidbMonitor` CR is as follows:
```

File: zh/aggregate-multiple-cluster-monitor-data.md [hunk 5]
```diff
@@ -11,19 +11,19 @@ summary: 通过 Thanos 框架聚合多个 TiDB 集群的监控数据
 
 Thanos 是 Prometheus 高可用的解决方案，用于简化 Prometheus 的可用性保证。详细内容请参考 [Thanos 官方文档](https://thanos.io/design.md/)。
 
-Thanos 提供了跨 Prometheus 的统一查询方案 [Thanos Query](https://thanos.io/components/query.md/) 组件，我们可以利用这个功能解决 TiDB 多集群监控数据聚合的问题。
+Thanos 提供了跨 Prometheus 的统一查询方案 [Thanos Query](https://thanos.io/components/query.md/) 组件，可以利用这个功能解决 TiDB 多集群监控数据聚合的问题。
```
Comment (ran-huang) [inline zh/aggregate-multiple-cluster-monitor-data.md:14]: Good job :)

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
+---
+title: Aggregate Monitoring Data of Multiple TiDB Clusters
+summary: Learn how to aggregate monitoring data of multiple TiDB clusters by Thanos query.
+---
+
+# Aggregate Monitoring Data of Multiple TiDB Clusters
+
+This document describes how to aggregate the monitoring data of multiple TiDB clusters by Thanos to solve the centralization problem of monitoring data of multiple clusters.
```
[the remark this answers is not in the document]
Comment (DanielZhangQD) [reply en/aggregate-multiple-cluster-monitor-data.md:8]: Yes, it's good.

File: en/aggregate-multiple-cluster-monitor-data.md (hunk 1 above)
Comment (DanielZhangQD) [reply en/aggregate-multiple-cluster-monitor-data.md:18]: First, deploy the TidbMonitor with Thanos Sidecar container:

{{< copyable "shell-regular" >}}

```shell
kubectl -n ${namespace} apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/monitor-with-thanos/tidb-monitor.yaml"
```

Then deploy the Thanos Query component:

{{< copyable "shell-regular" >}}

```shell
kubectl -n ${namespace} apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/monitor-with-thanos/thanos-query.yaml"
```

In Thanos Query, a Prometheus corresponds to a Store, which corresponds to a TidbMonitor. After deploying Thanos Query, you can provide a unified query interface for monitoring data through Thanos Query's API.

@ran-huang @Joyinqin PTAL.

Comment (ran-huang) [reply en/aggregate-multiple-cluster-monitor-data.md]: @Joyinqin Can you add this to the Chinese doc? Thank you.

**Comment by Joyinqin** [comment]
@ran-huang PTAL again, thx~

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+kubectl -n ${namespace} apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/monitor-with-thanos/thanos-query.yaml"
+```
+
+In Thanos Query, a Prometheus corresponds to a Store, which corresponds to a TidbMonitor. After deploying Thanos Query, you can provide a unified query interface for monitoring data through Thanos Query's API.
+
+## Configure Grafana
+
+After you deploy Thanos Query, change Grafana's DataSource into Thanos to query the monitoring data of multiple `TidbMonitor` CRs.
+
+## Add or reduce TidbMonitor
```
Comment (ran-huang) [inline en/aggregate-multiple-cluster-monitor-data.md:40]:
```suggestion
## Add or remove TidbMonitor
```

**Review by ran-huang — APPROVED** [review]
Rest LGTM

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 8]
```diff
@@ -0,0 +1,86 @@
...
+
+If you need to update the `TidbMonitor` CR or take it offline, update the starting configuration `--store` of Thanos Query Store, and perform a rolling update to the Thanos Query component.
```
Comment (DanielZhangQD) [inline en/aggregate-multiple-cluster-monitor-data.md:42]: `If you need to update the `TidbMonitor` CR or take it offline`, the original meaning is to add, update, or remove a monitor store from the Thanos query, the configuration is similar with the following:
```
$ thanos query \
    --http-address     "0.0.0.0:9090" \
    --store            "<store-api>:<grpc-port>" \
    --store            "<store-api2>:<grpc-port>" 
```

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 9]
```diff
@@ -0,0 +1,86 @@
...
+
+In Thanos Query, a Prometheus corresponds to a Store, which corresponds to a TidbMonitor. After deploying Thanos Query, you can provide a unified query interface for monitoring data through Thanos Query's API.
+
+## Configure Grafana
+
+After you deploy Thanos Query, change Grafana's DataSource into Thanos to query the monitoring data of multiple `TidbMonitor` CRs.
+
+## Add or remove TidbMonitor
+
+If you need to update the `TidbMonitor` CR or take it offline, update the starting configuration `--store` of Thanos Query Store, and perform a rolling update to the Thanos Query component.
```
Comment (ran-huang) [inline en/aggregate-multiple-cluster-monitor-data.md:42]:
```suggestion
If you need to add, update, or remove a monitor store from the Thanos query, update the configuration as follows and perform a rolling update to the Thanos Query component:

        ```shell
        thanos query \
            --http-address     "0.0.0.0:9090" \
            --store            "<store-api>:<grpc-port>" \
            --store            "<store-api2>:<grpc-port>" 
        ```

```
@DanielZhangQD Is this what you mean?

Comment (DanielZhangQD) [reply en/aggregate-multiple-cluster-monitor-data.md:42]:
```suggestion
If you need to add, update, or remove a monitor store from the Thanos query, update the `--store` configuration of Thanos Query, and perform a rolling update to the Thanos Query component.
```
What about this?

Comment (ran-huang) [reply en/aggregate-multiple-cluster-monitor-data.md:42]: LGTM. @Joyinqin Can you update Chinese as well?

**Comment by ran-huang** [comment]
@DanielZhangQD PTAL at a192fa5

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+Thanos provides [Thanos Query](https://thanos.io/tip/components/query.md/) component as a unified query solution across multiple Prometheus clusters. You can use this feature to aggregate monitoring data of multiple TiDB clusters.
+
+## Configure Thanos Query
+
+First, deploy the TidbMonitor with Thanos Sidecar container:
+
+{{< copyable "shell-regular" >}}
+
+```
+kubectl -n ${namespace} apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/monitor-with-thanos/tidb-monitor.yaml"
```
Comment (DanielZhangQD) [inline en/aggregate-multiple-cluster-monitor-data.md:23]:
```suggestion
kubectl -n ${namespace} apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/monitor-with-thanos/tidb-monitor.yaml
```

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+```
+
+Then deploy the Thanos Query component:
+
+{{< copyable "shell-regular" >}}
+
+```
+kubectl -n ${namespace} apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/monitor-with-thanos/thanos-query.yaml"
```
Comment (DanielZhangQD) [inline en/aggregate-multiple-cluster-monitor-data.md:31]:
```suggestion
kubectl -n ${namespace} apply -f https://raw.githubusercontent.com/pingcap/tidb-operator/master/examples/monitor-with-thanos/thanos-query.yaml
```

File: en/aggregate-multiple-cluster-monitor-data.md [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+
+## Add or remove TidbMonitor
+
+If you need to add, update, or remove a monitor store from the Thanos Query, update the `--store` configuration of Thanos Query, and perform a rolling update to the Thanos Query component.
+
+```shell
+        thanos query \
+            --http-address     "0.0.0.0:9090" \
+            --store            "<store-api>:<grpc-port>" \
+            --store            "<store-api2>:<grpc-port>"
```
Comment (DanielZhangQD) [inline en/aggregate-multiple-cluster-monitor-data.md:48]:
```suggestion
- query
- --http-address=0.0.0.0:9090
- --store=<store-api>:<grpc-port>
- --store=<store-api2>:<grpc-port>
```

**Comment by ran-huang** [comment]
@Joyinqin Please address comments. Thanks!

**Pull request merged**